lib/large_object_store.rb

```diff
@@ -19,7 +20,20 @@ def initialize(store)
 
     def write(key, value, options = {})
       value = Marshal.dump(value)
-      value = Zlib::Deflate.deflate(value) if options.delete(:compress)
+
+      compress_limit = options[:compress_limit] || DEFAULT_COMPRESS_LIMIT
```

Comment:
I get what you're saying... but I was copying the [Rails behavior](https://github.com/rails/rails/blob/6c61ae816ad7326e510a8f019cadc6db4cc9dae0/activesupport/lib/active_support/cache.rb#L601-L617) for deciding whether to compress or not. I think the idea is, you can pass `compress: true` if you're interested in compressing large objects but you don't have to decide at the call site whether the object is large or not.

`compress: true` in Rails kind of means "compress large objects", not "compress"